protocol-tests/src/test/java/fixtures/specialheader/implementation/HeadersImpl.java

```diff
@@ -0,0 +1,596 @@
+    }
+
+    /**
+     * The interface defining all the services for SpecialHeaderHeaders to be used by the proxy service to perform REST
+     * calls.
+     */
+    @Host("{$host}")
+    @ServiceInterface(name = "SpecialHeaderHeaders")
+    private interface HeadersService {
+        @Post("/status/500")
+        @ExpectedResponses({200})
+        Mono<Response<BinaryData>> paramRepeatabilityRequest(
+                @HostParam("$host") String host, RequestOptions requestOptions, Context context);
+
+        @Get("/specialHeader/repeatabilityRequestGet")
+        @ExpectedResponses({200})
+        Mono<Response<BinaryData>> paramRepeatabilityRequestGet(
+                @HostParam("$host") String host, RequestOptions requestOptions, Context context);
+
+        @Post("/specialHeader/repeatabilityRequestLRO")
+        @ExpectedResponses({200, 202})
+        Mono<Response<BinaryData>> paramRepeatabilityRequestLro(
+                @HostParam("$host") String host, RequestOptions requestOptions, Context context);
+
+        @Post("/specialHeader/repeatabilityRequestPageable")
+        @ExpectedResponses({200})
+        Mono<Response<BinaryData>> paramRepeatabilityRequestPageable(
+                @HostParam("$host") String host, RequestOptions requestOptions, Context context);
+
+        @Get("{nextLink}")
+        @ExpectedResponses({200})
+        Mono<Response<BinaryData>> paramRepeatabilityRequestPageableNext(
+                @PathParam(value = "nextLink", encoded = true) String nextLink,
+                @HostParam("$host") String host,
+                RequestOptions requestOptions,
+                Context context);
+    }
+
+    /**
+     * Send a post request with header Repeatability-Request-ID and Repeatability-First-Sent.
+     *
+     * <p><strong>Response Body Schema</strong>
+     *
+     * <pre>{@code
+     * Object
+     * }</pre>
+     *
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @return any object along with {@link Response} on successful completion of {@link Mono}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<BinaryData>> paramRepeatabilityRequestWithResponseAsync(RequestOptions requestOptions) {
+        RequestOptions requestOptionsLocal = requestOptions == null ? new RequestOptions() : requestOptions;
+        requestOptionsLocal.setHeader("Repeatability-Request-ID", UUID.randomUUID().toString());
+        requestOptionsLocal.setHeader(
+                "Repeatability-First-Sent",
+                DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss z", Locale.ENGLISH)
+                        .withZone(ZoneId.of("GMT"))
+                        .format(OffsetDateTime.now()));
+        return FluxUtil.withContext(
+                context -> service.paramRepeatabilityRequest(this.client.getHost(), requestOptionsLocal, context));
+    }
```

Comment:
Generated code.

`DateTimeFormatter` might not be efficient. Will consider util in azure-core if this pattern is widely used.

Reply:
One problem of the `setHeader` is that it might overwrite user setting. And user would need to use the `Context` to overwrite this `setHeader`.

Should we switch to `addRequestCallback` (but this is lots of stupid code)?
```
        requestOptionsLocal.addRequestCallback(request -> {
            if (request.getHeaders().getValue("Repeatability-Request-ID") == null) {
                request.getHeaders().set("Repeatability-Request-ID", UUID.randomUUID().toString());
            }
        });
```